week3/main.cpp

```diff
@@ -0,0 +1,139 @@
+//CodeU: Assignment_3
+//
+//  Created by Julia Semavina on 8.06.2017
+//
+//  problem specification:
+//      Given a grid of letters and a dictionary, find all the words from the dictionary that can be
+//      formed in the grid.
+//
+//  idea:
+//      We start building words from every cell of the grid. So, at first we have words of the length=1. If the word
+//      is prefix, we add one letter (from the North, South, East, West, N-W, N-E, S-W, S-E) if it is possible. Now we
+//      have words of the length=2.
+//      But this way if the input grid consists of the same items and input dictionary contains a very long string
```

Comment:
C is some constant (8 in this case)
Yes, in the worst case MaxLength is SizeOfGrid, but it won't be C^MaxLength. I don't know what it will be exactly, but in a simple case like when the grid contains the same letters it will be just O(SizeOfLength).